src/field_effect.c

```diff
@@ -3226,7 +3228,8 @@ static void FlyOutFieldEffect_ShowMon(struct Task *task)
     {
         task->tState++;
         gFieldEffectArguments[0] = task->tMonId;
-        FieldEffectStart(FLDEFF_FIELD_MOVE_SHOW_MON_INIT);
+        if (!FlagGet(FLAG_SYS_POKENAV_FLY))
+            FieldEffectStart(FLDEFF_FIELD_MOVE_SHOW_MON_INIT);
```

Comment:
This is going to conflict with fly the move right if the flag is set?

Reply:
it shouldn't, as the flag is only set when flying from the pokenav, and cleared when the FlyIn animation starts.